workflows/metatdenovo.nf

```diff
@@ -179,12 +189,26 @@ workflow METATDENOVO {
     PROKKA(MEGAHIT_INTERLEAVED.out.contigs.splitFasta( size: 10.MB, file: true).map { [[id: 'part_of_all_samples'], it] }, [], [] )
     ch_versions = ch_versions.mix(PROKKA.out.versions)
 
-//    //
-//    // MODULE: Call Prodigal
-//    //
-//    PRODIGAL([ [id: 'full_assembly' ], MEGAHIT_INTERLEAVED.out.contigs)
-//    ch_versions = ch_versions.mix(PRODIGAL.out.versions)
-
+    //
+    // MODULE: Call Prodigal
+    //
+    UNPIGZ_MEGAHIT_CONTIGS(ch_assembly_contigs)
+    //ch_assembly_contigs_unzipped = UNPIGZ_MEGAHIT_CONTIGS.out.gunzip
+    ch_versions = ch_versions.mix(UNPIGZ_MEGAHIT_CONTIGS.out.versions)
+    
+    ch_prodigal = Channel.empty()
+    if(params.orf_caller == 'prodigal'){
```

Comment:
So, constants were not used in the example code?

Reply:
The module works without constants. It is easier to write.

Reply:
It's not the module that should have a constant, it's the if clause that should maybe look like this:
```
if ( params.orf_caller == ORF_CALLER_PRODIGAL ) {
```
With `ORF_CALLER_PRODIGAL` declared somewhere. This is good practice in most programming languages, but we can live without that for the moment.

Reply:
I changed this line so now it has ORF_CALLER_PRODIGAL in it